### PR TITLE
EDSC-3028: Focused temporal

### PR DESCRIPTION
**Pull request by trevorlang**
Allows a user to focus a temporal interval

### CODE REVIEW

File: src/components/TimelineInterval/TimelineInterval.js [hunk 1]
```diff
@@ -16,11 +17,14 @@ import './TimelineInterval.scss'
  * @param {Integer} param0.zoomLevel Current zoom level of the timeline
  */
 export const TimelineInterval = ({
+  focusable,
```
Comment (macrouch) [inline src/components/TimelineInterval/TimelineInterval.js:23]: Add comments for new props

File: src/components/TimelineList/TimelineList.js [hunk 2]
```diff
@@ -35,10 +42,12 @@ export const TimelineList = ({
   timelinePosition,
   timelineWrapperRef,
   zoomLevel,
+  onFocusedClick,
   onTemporalMarkerMouseDown,
   onTimelineMouseDown,
   onTimelineMouseMove
 }) => {
+  // console.log('🚀 ~ file: TimelineList.js ~ line 50 ~ timelinePosition', timelinePosition)
```
Comment (macrouch) [inline src/components/TimelineList/TimelineList.js]: remove

File: src/constants.js [hunk 3]
```diff
@@ -35,3 +35,6 @@ export const ZOOM_LEVELS = {
 
 // The height in px of the area that will trigger a temporal selection
 export const TEMPORAL_SELECTION_HEIGHT = 20
+
+// The height in px of the area that will trigger a focused date selection
+export const FOCUSED_SELECTION_HEIGHT = 24
```
Comment (macrouch) [inline src/constants.js]: Remove

File: src/index.scss [hunk 4]
```diff
@@ -42,7 +47,7 @@
   }
 
   &__center {
-    display: none;
+    // display: none;
```
Comment (macrouch) [inline src/index.scss]: remove

File: src/index.js [hunk 5]
```diff
@@ -14,12 +13,14 @@ import { calculateTimeIntervals } from './utils/calculateTimeIntervals'
...
 import { generateEndTime } from './utils/generateEndTime'
 import { getCenterTimestamp } from './utils/getCenterTimestamp'
+import { getIntervalBounds } from './utils/getIntervalBounds'
 import { getIntervalsDuration } from './utils/getIntervalsDuration'
 import { getPositionByTimestamp } from './utils/getPositionByTimestamp'
 import { getTimestampByPosition } from './utils/getTimestampByPosition'
 import { roundTime } from './utils/roundTime'
 
 import {
+  FOCUSED_SELECTION_HEIGHT,
```
Comment (macrouch) [inline src/index.js]: remove

File: src/index.js [hunk 6]
```diff
@@ -494,6 +570,60 @@ export const EDSCTimeline = ({
...
+      zoomLevel
+    })
+
+    // Find the beginning and end of the interval where the timestamp sits
+    const {
+      end: newEnd,
+      start: newStart
+    } = getIntervalBounds(timeIntervals, timestamp)
+
+    // If the focused interval is outside of the temporal bounds, return
```
Comment (macrouch) [inline src/index.js]: Move this logic to TimelineInterval

File: src/index.js [hunk 7]
```diff
@@ -557,6 +686,94 @@ export const EDSCTimeline = ({
...
+    if (temporalStart && temporalEnd) {
+      if (direction === 'previous' && temporalStart >= currentStart) return
+      if (direction === 'next' && temporalEnd <= currentEnd) return
+    }
+
+    // `end` is within the intervals of timeIntervals, use it with getIntervalBounds to find the bounds that include it
+    const newFocused = getIntervalBounds(timeIntervals, currentEnd, delta)
+
+    // Update the focusedInterval state
+    // call onFocusedSet
```
Comment (macrouch) [inline src/index.js]: remove

File: src/index.js [hunk 8]
```diff
@@ -557,6 +686,94 @@ export const EDSCTimeline = ({
...
+      timeIntervals,
+      timelineToolsWidth,
+      zoomLevel,
+      wrapperWidth: timelineWrapperWidth
+    })
+
+    // Find the offset of the current timeline center and the current focused interval start
+    const offset = currentCenterTime - currentStart
+
+    // this needs to be the width of an interval, not the offset from center, the scroll methods are only looking at the left position and keeping that
```
Comment (macrouch) [inline src/index.js]: reword comment

**Review by macrouch — APPROVED** [review]

**Review by abbottry — APPROVED** [review]

**Pull request merged**